app/server_test.go

```diff
@@ -565,3 +566,11 @@ func TestSentry(t *testing.T) {
 		}
 	})
 }
+
+func TestCancelTaskSetsTaskToNil(t *testing.T) {
+	var taskMut sync.Mutex
+	task := model.CreateRecurringTaskFromNextIntervalTime("a test task", func() {}, 5*time.Minute)
+	require.NotNil(t, task)
+	cancelTask(&taskMut, &task)
+	require.Nil(t, task)
```

Comment:
Could we call cancelTask here again, which was the cause of the problem?

Reply:
Done